.github/build-docker-images.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+# SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+REPO=tenstorrent/tt-mlir
+BASE_IMAGE_NAME=ghcr.io/$REPO/tt-mlir-base-ubuntu-22-04
+CI_IMAGE_NAME=ghcr.io/$REPO/tt-mlir-ci-ubuntu-22-04
+BASE_IRD_IMAGE_NAME=ghcr.io/$REPO/tt-mlir-base-ird-ubuntu-22-04
+IRD_IMAGE_NAME=ghcr.io/$REPO/tt-mlir-ird-ubuntu-22-04
+
+# Compute the hash of the Dockerfile
+DOCKER_TAG=$(./.github/get-docker-tag.sh)
+echo "Docker tag: $DOCKER_TAG"
+
+# Get the SHA of the current commit
+CURRENT_SHA=$(git rev-parse HEAD)
+echo "CURRENT_SHA: $CURRENT_SHA"
+
+# Are we on main branch
+ON_MAIN=$(git branch --show-current | grep -q main && echo "true" || echo "false")
+
+build_and_push() {
+    local image_name=$1
+    local dockerfile=$2
+    local on_main=$3
+    local from_image=$4
+
+
+    if docker manifest inspect $image_name:$DOCKER_TAG > /dev/null; then
+        echo "Image $image_name:$DOCKER_TAG already exists"
+    else
+        echo "Building image $image_name:$DOCKER_TAG"
+        docker build \
+            --progress=plain \
+            --build-arg GIT_SHA=$CURRENT_SHA \
+            --build-arg FROM_TAG=$DOCKER_TAG \
+            ${from_image:+--build-arg FROM_IMAGE=$from_image} \
+            -t $image_name:$DOCKER_TAG \
+            -t $image_name:latest \
+            -f $dockerfile .
+
+        echo "Pushing image $image_name:$DOCKER_TAG"
+        docker push $image_name:$DOCKER_TAG
```

Comment:
btw where does this push? Is it local to our runners or to some server somewhere?

Reply:
It depends on your Docker config. 
If it is run in Github workflow it will push the image to ghcr registry (login is in workflow). 
You can also run it locally and it will push to wherever your docker config points to.